openassessment/xblock/utils/xml.py

```diff
@@ -772,6 +772,12 @@ def serialize_content_to_xml(oa_block, root):
     if oa_block.show_rubric_during_response is not None:
         root.set('show_rubric_during_response', str(oa_block.show_rubric_during_response))
 
+    # Serialize and add tag data if any
+    if hasattr(oa_block, 'serialize_tag_data') and callable(oa_block.serialize_tag_data):  # pylint: disable=no-member
+        tag_data = oa_block.serialize_tag_data()  # pylint: disable=no-member
+        if tag_data:
+            root.set('tags-v1', tag_data)
```

Comment:
I suggest that in `TaggedBlockMixin` you put these lines of code:

https://github.com/openedx/edx-platform/blob/2cd2f09da1a7cf1a609ca1309a35850458a166c7/cms/lib/xblock/tagging/tagged_block_mixin.py#L51-L54

into a helper function called `add_tags_to_node`. And then this can just become:

```suggestion
    if hasattr(oa_block, 'add_tags_to_node') and callable(oa_block.add_tags_to_node):  # pylint: disable=no-member
        # This comes from TaggedBlockMixin
        oa_block.add_tags_to_node(root)  # pylint: disable=no-member
```

Reply:
Agreed, much cleaner. I've updated it, and also updated the edx-platform PR to use it as well.